tests/rptest/tests/controller_upgrade_test.py

```diff
@@ -63,6 +63,13 @@ def test_updating_cluster_when_executing_operations(self):
         else:
             admin_operations = [o for o in RedpandaAdminOperation]
 
+        # DeleteRecords is not an allowable admin operation until the cluster
+        # has been fully upgraded
+        admin_operations = [
+            o for o in admin_operations
+            if o != RedpandaAdminOperation.DELETE_RECORDS
+        ]
+
```

Comment:
Maybe we could only do this if prev_version is less than 23.2. Then we don't have to update this later on